### PR TITLE
Add lsp-erlang-server-connection-type custom var to specify between 'tcp and 'stdio

### PR DESCRIPTION
**Pull request by jfacorro**
erlang-ls/erlang_ls#16

### CODE REVIEW

**Comment by jfacorro** [comment]
@robertoaloi

File: lsp-erlang.el [hunk 1]
```diff
@@ -38,12 +38,25 @@
...
+  :risky t
+  :type 'string)
+
 (defun lsp-erlang-server-start-fun (port)
   `(,lsp-erlang-server-path
-    ,(number-to-string port)))
+    "--transport" "tcp"
+    "--port" ,(number-to-string port)))
+
+(defun lsp-erlang-server-connection ()
```
Comment (yyoncho) [inline lsp-erlang.el]: Here you could return a function that will return the function to create the connection based on the connection-type.

Comment (yyoncho) [reply lsp-erlang.el]: (not necessary to address).

**Comment by yyoncho** [comment]
Looks good to me, I will wait for @robertoaloi to review.

File: lsp-erlang.el [hunk 2]
```diff
@@ -38,12 +38,25 @@
   :risky t
   :type 'file)
 
+(defcustom lsp-erlang-server-connection-type
+  'tcp
+  "Type of connection to use with the Erlang Language Server: tcp or stdio"
+  :group 'lsp-erlang
+  :risky t
+  :type 'string)
```
Comment (stormwatch) [inline lsp-erlang.el:46]: type should be a symbol, not a string.

**Pull request merged**